synapse/datamodel.py

```diff
@@ -64,14 +70,31 @@ def onSet(self, func):
         Args:
             func (function): A prop set callback.
 
-        The callback is called with the current transaction,
-        the node, and the old property value (or None).
+        The callback is called within the current transaction,
+        with the node, and the old property value (or None).
 
-        def func(xact, node, oldv):
+        def func(node, oldv):
             dostuff()
         '''
         self.onsets.append(func)
 
+    def onDel(self, func):
+        '''
+        Add a callback for deleting this property.
+
+        The callback is executed after the property is deleted.
+
+        Args:
+            func (function): A prop del callback.
+
+        The callback is called within the current transaction,
+        with the node, and the old property value (or None).
+
+        def func(node, oldv):
+            dostuff()
+        '''
+        self.ondels.append(func)
```

Comment:
What if you add the same function multiple times? Sort of a silly thing to do, but it would probably call it multiple times.